src/coap_client.c

```diff
@@ -261,6 +261,34 @@ static enum golioth_status golioth_coap_client_set_internal(struct golioth_clien
     return GOLIOTH_OK;
 }
 
+enum golioth_status golioth_coap_client_post(struct golioth_client *client,
+                                             const char *path_prefix,
+                                             const char *path,
+                                             enum golioth_content_type content_type,
+                                             const uint8_t *payload,
+                                             size_t payload_size,
+                                             golioth_post_cb_fn callback,
+                                             void *callback_arg,
+                                             bool is_synchronous,
+                                             int32_t timeout_s)
+{
+    struct golioth_coap_post_params params = {
+        .content_type = content_type,
+        .callback_post = callback,
+        .arg = callback_arg,
+        .callback_is_post = true,
```

Comment:
Perhaps not relevant to this PR, but this `callback_is_post` member makes me think of the `is_observe` member we have in some other params structs. Perhaps these could be unified in a `type` member that all params structs have with an enum that specifies is_post, is_observe, etc.

Reply:
Sure, I would 100% agree if we would have them in the same structure. Do you think it makes sense to introduce type now even if we have 2 types at most for each struct?

Even if we would switch to types, then I think that `observe` and `post` should be defined in separate enum, i.e. separate namespace. They are just different things, so I would prefer to not put them in single location to prevent confusion.